src/pages/workspace/members/WorkspaceInviteMessageComponent.tsx

```diff
@@ -232,21 +238,25 @@ function WorkspaceInviteMessageComponent({
                     </View>
                     <View style={[styles.mb3]}>
                         <View style={[styles.mhn5, styles.mb3]}>
-                            {isInviteNewMemberStep && (() => {
-                                const invitingMemberEmail = Object.keys(invitedEmailsToAccountIDsDraft ?? {}).at(0) ?? '';
-                                const memberDetails = getPersonalDetailByEmail(invitingMemberEmail);
-                                const memberName = Str.removeSMSDomain(memberDetails?.displayName ?? '');
-                                return (
-                                    <MenuItemWithTopDescription
-                                        title={memberName && memberName !== invitingMemberEmail ? memberName : invitingMemberEmail}
-                                        description={translate('common.member')}
-                                        shouldShowRightIcon
-                                        onPress={() => {
-                                            Navigation.goBack(backTo);
-                                        }}
-                                    />
-                                );
-                            })()}
+                        {isInviteNewMemberStep && (
+                            <MenuItemWithTopDescription
+                                  title={invitingMemberName && invitingMemberName !== invitingMemberEmail ? invitingMemberName : invitingMemberEmail}
+                                  description={translate('common.member')}
+                                  shouldShowRightIcon
+                                  onPress={() => {
+                                      clearInviteDraft(policyID);
+                                      setAssignCardStepAndData({
+                                          currentStep: CONST.COMPANY_CARD.STEP.ASSIGNEE,
+                                          data: {
+                                              ...assignCard?.data,
+                                              invitingMemberEmail: undefined,
+                                              invitingMemberAccountID: undefined,
+                                          },
+                                          isEditing: false,
+                                      });
+                                  }}
```

Comment:
Please move this callback method to be declared outside of the `return()` block.